collect_app/src/main/java/org/odk/collect/android/activities/LocationActivity.java

```diff
@@ -0,0 +1,13 @@
+package org.odk.collect.android.activities;
+
+/**
+ * Wraps location-based activity classes GeoPointActivity, GeoPointMapActivity and
+ * GeoPointOsmMapActivity. Provides a method to sanitize location provider names so "gps" displays
+ * as "GPS" in user-facing messaging.
+ */
+
+public abstract class LocationActivity extends CollectAbstractActivity {
```

Comment:
I would rather create a util class instead of adding a new activity.

Reply:
But is it an activity?

Reply:
It extends activity so it's also an activity.

Reply:
It contains behavior common to one subset of CollectAbstractActivity. I don’t see it as we’re adding another activity. How do you see it that way?

Reply:
We’re not increasing the number of concrete Activity classes.

Reply:
I mean it's not a separate activity but it's another layer created just for one method what doesn't really makes sense to me.

Reply:
OK, I understand your objection.

How would creating a util class be an improvement? It too would be a class created just for one method.

Reply:
Yes but then in:
GeoPointActivity
GeoPointMapActivity
GeoPointOsmMapActivity

you don't need to extend that new class and you can use CollectAbstractActivity as it was what makes the code clearer. I know that maybe it doesn't go really well so far but we wanted to start introducing patterns like MVVM and separating such code from activities, so creating a new activity layer just for that method doesn't look good.

Reply:
I find your argument compelling that we shouldn’t add code like this to a class in the Activity hierarchy, because it not a concern of an activity. Thanks for explaining. 

What would you recommend for the name of a class to hold this new method, and in what package would you put it?

Reply:
the package is easy because we have one called org.odk.collect.android.utilities but when it comes to the name... maybe just `GeoPointUtils`

Reply:
Thanks, @grzesiek2010! @KiaFarhang, how does this sound to you?

Reply:
I feel like I just got a free Java best practices class by reading these comments :) Thanks very much @grzesiek2010 and @dcbriccetti - here's a list of the changes I'll make:

- Cut the abstract `LocationActivity` class in favor of a `GeoPointUtils` utility class under `org.odk.collect.android.utilities`
- Looking at other utility classes I'm seeing a fair number of static methods for issues like these when we just need to modify a String. I'll go with static for now, but changing would be easy so let me know if we should take another approach.
- I'll follow @dcbriccetti's recommendations on formatting, wording and capitalization in the method name.
- Will also throw a test class in for the utility class, with a couple quick unit tests for this capitalization method.

Thanks again to both of you for the help!